Sem. 01/task08.cpp

```diff
@@ -0,0 +1,138 @@
+#include <iostream>
+#include "Source.h"
+
+using std::cin;
+using std::cout;
+using std::endl;
+
+const unsigned int MAX_SIZE = 30;
+
+namespace StudentNs
+{
+	struct Student 
+	{
+		char name[MAX_SIZE];
+		char surname[MAX_SIZE];
+		double grade;
+	};
+
+	void createStudent(Student& student)
+	{
+		cin >> student.name >> student.surname >> student.grade;
+		//validateGrade(student.grade);
+	}
+}
+
+namespace GroupNs 
+{
+	struct Group 
+	{
+		int numStudents;
+		double averageGrade;
+		StudentNs :: Student* students;
+	};
+
+	void validateGrade(double a)
+	{
+		if (a < 2.0 && a > 6.0)
+		{
+			cout << -1;
+		}
+	}
+
+	double calculateAverageGroupScore(StudentNs::Student* students, const int& countStudents)
```

Comment:
const Student*, do not put & on primitive types - & takes between 4 and 8 bytes, which is more than any primitive (except the double in the case &=4bytes)